### PR TITLE
fix(surveys): Validate HTML for all questions and descriptions

### PR DESCRIPTION
**Pull request by neilkakkar**
## Changes

Part 1 of 2 of custom HTML support in surveys. This ensures parsed HTML is safe and valid to use
...

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers

### CODE REVIEW

**Comment by neilkakkar** [comment]
the cost of purifying & bundling together is a bigger package 😬

File: rollup.config.js [hunk 1]
```diff
@@ -56,7 +56,9 @@ export default [
                 name: 'posthog',
             },
         ],
-        plugins: [...plugins],
+        // We need to make sure DOMPurify is bundled together with the surveys code
+        // hence the no-module option resolution
+        plugins: [resolve(), ...plugins],
```
[the remark this answers is not in the document]
Comment (neilkakkar) [reply rollup.config.js:61]: Actually, the `resolve()` configuration options for other things doesn't _appear_ to make a difference, do you remember what it was for?

File: src/extensions/surveys.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ import {
     SurveyAppearance,
     SurveyQuestion,
 } from '../posthog-surveys-types'
+import DOMPurify from 'dompurify'
+// import { sanitize } from 'dompurify'
```
Comment (liyiy) [inline src/extensions/surveys.ts:12]: commented out code

**Review by liyiy — APPROVED** [review]

File: src/extensions/surveys.ts (hunk 2 above)
Comment (neilkakkar) [reply src/extensions/surveys.ts:12]: nice thanks

**Review by jurajmajerik — APPROVED** [review]

File: rollup.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjackwhite) [reply rollup.config.js:61]: No idea 🙈

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Personally I am not a fan of this. 22kb of extra JS just to make sure the html is valid and pure which _could_ instead be done at write time instead? 

Is there a reason why we aren't doing this in app.posthog.com instead?

**Comment by neilkakkar** [comment]
Discussed offline, conclusion is @benjackwhite convinced me that my reluctance to do this validation on the backend is not worth the trade-offs. Bigger package size is annoying to users, plus the security concern of people being able to bypass our API isn't big enough to warrant sanitisation in the frontend. (if that can happen, we have much more serious issues to deal with)

**Comment by neilkakkar** [comment]
And at the same time, the feature to allow scripts / custom HTML rendering via survey API mode is also something we're not going to support. Since it's API mode, they can just add this to the code themselves, rather than parsing these scripts via the survey itself.

**Pull request closed without merging**